scripts/data_prep.py

```diff
@@ -0,0 +1,59 @@
+import pandas as pd  # type: ignore
+
+from sklearn.model_selection import train_test_split  # type: ignore
+
+from pathlib import Path
+import argparse
+import logging
+
+
+def clean_dataset(df: pd.DataFrame) -> pd.DataFrame:
+    clean_df = df.copy(deep=True)
+    clean_df.loc[clean_df["title"].str.split().str.len() == 0, "title"] = (
+        pd.NA
+    )  # Empty title is same as NaN
+    clean_df.loc[clean_df["text"].str.split().str.len() == 0, "text"] = (
+        pd.NA
+    )  # Empty text is same as NaN
+    clean_df = clean_df.dropna(axis=0, how="any")
+    return clean_df
+
+
+def preprocess_dataset(
+    orig_df: pd.DataFrame, test_size: float
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    logging.info("Cleaning data...")
+    clean_df = clean_dataset(orig_df)
+
+    logging.info("Train/test splitting data...")
+    train_df, test_df = train_test_split(clean_df, test_size=test_size, random_state=42)
```

Comment:
Maybe we should also add `random_state` as a parameter, so that we can control the seed globally from e,g. some config file